test/src/Makefile.am

```diff
@@ -51,20 +51,18 @@ CIOD_INC_FLAGS = -I@ARCHHEADER@ \
 endif
 
 if WITH_TEST_INSTALLED
```

Comment:
So, what happens if you don't specify `API_DIR` and `API_LIB_DIR` in this `WITH_TEST_INSTALLED` automake conditional? How does this pick up the "right" `API_*` macroes? 

Also, did you check the rpath for "installed test programs"? If they are dynamically linked to the LMON libraries, They should point to the installed LMON API libraries, not the libraries from the build directory... I think.

Reply:
similar to my reply to your other comment, this is why I pulled the definitions out of the if and else branches and made the definition unconditional. The rpaths do indeed point to the installed libraries, not the build ones. I think the autotools magic takes care of this:


```
[lee218@rzwiz2:launchmon]$ head config.log 
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by LaunchMON configure v1.0.2-38-gb659820, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./configure --prefix=/collab/usr/global/tools/launchmon/toss_3_x86_64_ib/launchmon-test2/ --with-test-rm=slurm --with-test-ncore-per-CN=2 --with-test-nnodes=1 --with-test-rm-launcher=/usr/bin/srun --with-test-installed=install-tests

## --------- ##
## Platform. ##

[lee218@rzwiz2:launchmon]$ pwd
/g/g0/lee218/src/launchmon

[lee218@rzwiz2:launchmon]$ readelf -d /collab/usr/global/tools/launchmon/toss_3_x86_64_ib/launchmon-test2/share/launchmon/tests/be_kicker | grep RPATH
 0x000000000000000f (RPATH)              Library rpath: [/usr/tce/packages/gcc/gcc-4.9.3/lib:/usr/tce/packages/gcc/gcc-4.9.3/lib64:/collab/usr/global/tools/launchmon/toss_3_x86_64_ib/launchmon-test2/lib:/usr/tce/packages/gcc/gcc-4.9.3/lib64/../lib64]

```

Reply:
OK sounds good.